posts/2022-10-25-22.0.0.11.adoc

```diff
@@ -0,0 +1,440 @@
+
+[source, xml]
+----
+<featureManager>
+    <feature>appSecurity-3.0</feature>
+</featureManager>
+
+<!-- 
+    The 3rd-party JCache provider library that Liberty will use to manage and connect to the cache.
+ -->
+<library id="JCacheProviderLib">
+    <fileset dir="${shared.resource.dir}" includes="jcacheprovider.jar" />
+</library>
+
+<!-- 
+    This shared library contains any custom credentials and/or principals that
+    are stored in the subject.
+ -->
+<library id="CustomLib">
+    <fileset dir="${shared.resource.dir}" includes="customlibrary.jar" />
+</library>
+
+<cache ... >
+    <cacheManager ... >
+        <cachingProvider jCacheLibraryRef="JCacheProviderLib" commonLibraryRef="CustomLib" />
+    </cacheManager>
+</cache>
+
+<!--
+    Some sample JAAS custom login module configuration. The custom login module
+    in this example would inject custom credentials or principals into the subject.
+
+    Note that the 'libraryRef' in the 'jaasLoginModule' needs to be set to the same
+    library referenced from the caching provider.
+ -->
+<jaasLoginContextEntry id="system.WEB_INBOUND"
+    name="system.WEB_INBOUND"
+    loginModuleRef="custom, hashtable, userNameAndPassword, certificate, token" />
+
+<jaasLoginModule id="custom"
+    className="org.acme.CustomLoginModule"
+    controlFlag="REQUIRED" libraryRef="CustomLib" />
+
+<!-- 
+    Any applications that will be accessing classes from the Subject also need
+    to use the same library reference.
+ -->
+<application ...>
+    <classloader commonLibraryRef="CustomLib" />
+</application>
+----
+
+A few points to consider when configuring a JCache for use with the authentication cache.
+
+* The distributed authentication cache is comprised of keys and values of type `Object`.
+To match the behavior of the local authentication cache, set a least recently used eviction (`LRU`) policy with a maximum entry count of 25000 and an entry TTL of 600 seconds. Note that with distributed caches, partitioning of the cache can lead to an actual capacity below the configured value.
+* If your `JCache` provider supports it, configure a client-side cache to reduce transactions to the distributed cache. If the client-side cache supports storing the entries as deserialized objects, this can further improve performance.
```

Comment:
```suggestion
* If your JCache provider supports it, configure a client-side cache to reduce transactions to the distributed cache. If the client-side cache supports storing the entries as deserialized objects, this can further improve performance.
```